examples/github-complete/README.md

```diff
@@ -1,14 +1,29 @@
-# Complete Atlantis example with GitHub Webhooks
+# Complete Atlantis example with GitHub App and Webhooks
 
 Configuration in this directory creates the necessary infrastructure and resources for running Atlantis on Fargate plus GitHub repository webhooks configured to Atlantis URL.
 
 An existing Route53 hosted zone and domain is required to deploy this example.
 
-GitHub's personal access token can be generated at https://github.com/settings/tokens
+The GitHub App can be generated using multiple methods:
+
+- You can follow Atlantis instructions depicted [here](https://www.runatlantis.io/docs/access-credentials.html#github-app). The Atlantis method mostly automates the GitHub App generation using [GitHub App Manifest](https://docs.github.com/en/developers/apps/building-github-apps/creating-a-github-app-from-a-manifest), but you need an exposed endpoint to complete the process.
```

Comment:
This option doesn't seem possible because as you've mentioned you need an exposed endpoint to be able to setup the GitHub App. How would one go about doing that? Should I - 
 * Spin up a version of the `github-complete` example without a valid `github_app_id`, `github_app_key`, and `github_webhook_secret`
 * Register the GitHub app via the exposed endpoint
 * Re-run terraform with the GitHub app values?

More detail here is needed if this is the preferred path.

Reply:
I understand your pain... it's not very easy to bootstrap Atlantis using the official documentation.

What the Atlantis team did, was to try to ease the process of setting up the GitHub App by using the GitHub App manifest configuration method - this 'manifest' method, involves the Atlantis runtime to provide all the requirements for GitHub to create the app on Atlantis behalf. Of course, this raises security concerns: 
- GitHub needs to trust the exposed endpoint (this means not only being exposed, but needs to have a valid SSL certificate)
- You need to expose your app to the internet and follow the 'one-time' setup procedure to get the required secrets `github_app_id`, `github_app_key`, and `github_webhook_secret`.
- This method not only provides you the secrets, but also configures the GitHub App with the required permissions as depicted by the Atlantis team.

**TL;DR:** The steps you mentioned are correct. That's what you need to do in order to setup the GitHub App according to Atlantis official documentation.